### PR TITLE
Automotive

### PR DESCRIPTION
**Pull request by mfhepp**
Dear all:

This is the new automotive extension proposal, initially based on https://www.w3.org/wiki/WebSchemas/Vehicles. This pull request addresses the requests to 

2. separate the automotive part from the property-values proposal,
3. remove the bicycles, boats, etc. elements until there will be fully-fledged proposals for those domains and
4. various detailed comments from Tom March from Jan 9, 2015.

In the following, I summarize how these comments and requests have been addressed.

"In general, I think we should review and make changes to schema.org in smaller self-contained units. In this case, that would mean splitting out the changes into:
(1) the value-prop changes, which I will send a separate mail on.
(2) the auto changes
(3) the bicycle changes
(4) the boats changes"

--> OK, done.

"For the bicycle and boat changes, I think we should defer the changes for now. In both cases, I did a quick survey of boats/bicycles for sale and manufacturer sites, and I saw little overlap between the main attributes they used to describe the boats/bicycles vs. what is present in the Bicycle, Watercraft, Boat, MotorBoat, and SailingBoat proposals. I think we should revisit this as a separate change once there are more complete and detailed proposals for these types with their own use-cases, etc. (the use cases in the current proposal are all auto-specific)."

--> OK, done: removed Bicycle, Boat, Watercraft, Motorboat, Sailing Boat and removed/updated domain/range definitions.

"One small note on Bicycle, when we add it: why does Bicycle allow for “some bicycles have a small combustion or electric engine that assists with pedaling” when there is a separate MotorizedBicycle type for that?"

--> The idea was that, at least in Europe, there are MotorizedBicycles, which are essentially very light Motorcycles (their pedals are not typically used, except for starting the engine), and bikes with a very small support engine, like e-bikes. Since we removed all bikes parts for the moment, this issue is closed.

"Car vs. Van vs. Truck – Can this be modeled as body styles (now called bodyType) instead? The three example sites seem to do this, and it strikes me as cleaner from a vocabulary point of view, especially as companies continue to blur the lines between these (SUVs, crossovers, etc.)."

--> OK, done: Van and Truck have been removed. This will also make it easier for used car listing sites that may not have proper classification information in their DBs. Also note that Wikipedia has recently changed its definition for car to to "A car is a wheeled, self-powered motor vehicle used for transportation." I copied that. I also added an acknowledgment to Wikipedia to the GAO text, to give proper credit that some class and property definitions are inspired by Wikipedia.

So we now have the following subtypes of Vehicle:

MotorizedRoadVehicle
	BusOrCoach
	Car
	Motorcycle
	MotorizedBicycle

MotorizedRoadVehicle will be handy for non-cars, like trikes, and it also allows bundling all engine-related properties nicely.

"For EmissionStandardValue, TransmissionTypeValue, EngineTypeValue, and BodyTypeValue, there doesn’t seem to be any distinction in the usage vs. using Text, so why do we need these? If we were to have enumeration values for these, then having separate types would make more sense to me."

--> The initial idea was to to use / allow the use of URIs, like Freebase, DBPedia. 
DONE - I changed the range of all respective properties to Text OR QualitativeValue or URL and removed the classes for these types. That should cover all use-cases and reduce the size of the proposal.

"For FuelTypeValue, I saw on the US version of Autotrader (one of the examples cited on the wiki) values of Alternative, Diesel, Electric, Gasoline, and Hybrid. These are missing from the enumeration, and I think they should be added. Also, I haven’t seen a reference to TwoStrokeMixture (the only current enum value for this type) from the sorts of sites in the use case list before. Is that for motorcycles? Or if the entire purpose of this type is to add TwoStrokeMixture to the lists already available on DBPedia/Freebase/etc., then I would argue people can just enter text for that, and we can remove the FuelTypeValue in favor of having people directly use QualitativeValue."

--> DONE. I removed TwoStrokeMixture an the FuelType type,  and changed of all properties to Text or QualitativeValue or URL. This will allow standardization bodies to establish canonical URLs/URIs for important fuel types and foster the use of Wikipedia, DBPedia and Freebase IDs.

"For CarUsageType, the name and description don’t imply past usage to me. Should we rename this as something like PastCarUsageType instead? Otherwise, when I first read it, I was expecting this type to be used to describe the current use (i.e., available for rental now, not available for sale to a private owner but previously used as a rental car)."

--> Note that the property name is specialUsage, which I think fits. For the enumerations, I would like to keep CarUsageType, because we may want to reuse this for terms and conditions of rental offers or insurance contracts.
DONE: I changed the description of the values RentalUsage, DrivingSchoolUsage, TaxiUsage to prepare for such future usages but kept specialUsage for the property and CarUsageType for the enumerations.

"Can we rename ACRISSCode to acriss and VIN to vin? Other similar properties (isbn, gtin13, gtin14, etc.) follow this naming convention."
--> OK, DONE: Changed ACRISSCode to acrissCode and VIN to vin.

"Can we rename acceleration to accelerationTime? This would make the usage slightly clearer and also remove the issues in the notes about there not being unit codes for acceleration."

--> OK, DONE. 

"Can we rename colorInterior to interiorColor to be consistent with things like tongueWeight, seatingCapacity, modelDate, etc.?"

--> OK, DONE.

"How should a publisher specify the condition (new/used/etc.)? It seems like we could reuse the existing itemCondition property (with updated description) for this."

--> For the general condition, it will be itemCondition. For damages, there is the damages property.

"configurationName seems to be duplicating a bunch of other properties (transmission, enginePower, bodyType, etc.). If people are going to do this, IMO, they should just use the name property. There is a concept of trim level that the US version of AutoTrader uses, which I think is missing and would be useful. That would capture probably just the “ST 2.5” part of the example. For a Subaru Legacy, on the US AutoTrader site, for example, you’ll see values such as “2.5GT Limited”, “2.5i”, “3.6R Limited”, etc. "

--> I would like to keep that, because

1. the resulting strings may be very valuable for NLP processing (links to knowledge graphs);
2. I am thinking of an extension for modeling configurable products (cars and other), and for that I would like to have a generic name already; trim is very car-specific and does not even fit for other vehicles.
3. schema:name may include the full make and model, while configurationName will just contain strings referring to the car configuration. In European cars, this may be much more detailed than just trim information, because they are very highly configurable. The configuration string in there will be too long to be used as a short name.
4. Yandex initially requested this property, if I remember correctly.

"The engine properties (displacement, name, power, and type) should probably be separated out into their own type, especially if we want to model multiple engines per vehicle (as some of the descriptions suggest). Otherwise, it will not be explicit which property sets go together."

--> While we could add a type Engine and set the domain of all these properties to MotorizedRoadVehicle OR Engine, and define a property "engine" with a domain of MotorizedRoadVehicle and a range of Engine, I would like to keep that as a future extension, because most cars have only one engine, and the proposal is already pretty substantial.

"For enginePower, should we also list the unit code for horsepower (BHP?)?"

--> I decided against listing it, because horsepower is to my knowledge an obsolete unit. As far as I can see, the proper code would be N12 ("Pferdestaerke", with 1 PS = 735,498 75 W, btw. It could still be used, but I think we should not actively promote it.

"For fuelConsumption and fuelEfficiency, it would be good to have examples of these in addition to the existing notes. Also, I am not sure it makes sense to put the “wrong” units in the unitCode property. Why not guide people to use unitText instead? I thought that unitText was introduced for just this situation; i.e., when there is no standard code to refer to."

--> I fixed the description to recommend unitText. As for examples, I will put them on my radar, but I think such can be added in a second step.

"fuelType is on Vehicle, but fuelConsumption and fuelEfficiency are only on the more specific types. I think the placement of these three properties should be consistent, and I like putting them on the more specific types."

-> OK, DONE: I changed the domain of fuelType to MotorizedRoadVehicle. 

"For gears, the description says forward and reverse gears, but I think it is customary to only list the forward gears. The example for Car also seems to list only the forward gears. Should we update the description?"

--> Agreed. OK - DONE.

"For interiorType, if this is intended to describe just the material, should we rename it to interiorMaterial?"

--> I would like to keep interiorType, because it could also be a type that is not based on a material but on a target usage or audience ("Hunter edition with gun rack" or "1970s disco interior").

"It looks like numberOfOwners was renamed to previousOwners. It would be nice to have the wiki updated if so."

--> OK, DONE - but while I tried to sync the Wiki page with the github work, it is very resource-consuming and error-prone, so we should consider the Wiki page deprecated.

"What is the use case for productionDate? I’m not sure we need this property, but if we do, can we reuse the existing dateCreated property instead? It sounds like it would have the same meaning."

--> In Europe, the production date is often an essential characteristic for new and used car, because some cars stand on stock for several months depending on factory capacities and demand. The property was also present in the older Yandex proposal. So I suggest to keep that.

"Can we rename purchasingDate to purchaseDate and firstRegistration to firstRegisteredDate? Our date property naming in schema.org isn’t very self-consistent, but I think these suggestions will be more consistent."

--> OK, DONE: Renamed purchasingDate to purchaseDate and firstRegistration to dateOfFirstRegistration, which I think is better English.

"transmission mentions sending power to propellers, but this property is on MotorizedRoadVehicle. Should it instead be on Vehicle? Or should the description be updated?" 

--> OK, DONE: I changed the description so that it remains generic without irritating typical users.

"There are some typos:
In the description for fuelCapacity: staorage => storage"

--> OK, DONE.

"In the description for unitText: “Useful if you cannot provide a standard unit code for” what?"

--> This is an issue for https://github.com/mfhepp/schemaorg/tree/property-values. I will add it there.

"In the descriptions for payload and weightTotal: permited -> permitted"

-> OK, DONE. (Also found a few more).

I also updated the examples accordingly.

Again, a huge thanks for your detailed review! I hope this is ready for inclusion now.

PS: For tracking what I changed between the two version, see

    https://github.com/mfhepp/schemaorg/commit/ed235c80edfd60000e0d1b6cb58f04284f734ec1

Best wishes

Martin

### CODE REVIEW

**Comment by mfhepp** [comment]
This is https://github.com/schemaorg/schemaorg/issues/262 in the tracker.

**Comment by tmarshbing** [comment]
"configurationName seems to be duplicating a bunch of other properties (transmission, enginePower, bodyType, etc.). If people are going to do this, IMO, they should just use the name property. There is a concept of trim level that the US version of AutoTrader uses, which I think is missing and would be useful. That would capture probably just the “ST 2.5” part of the example. For a Subaru Legacy, on the US AutoTrader site, for example, you’ll see values such as “2.5GT Limited”, “2.5i”, “3.6R Limited”, etc. "
--> I would like to keep that, because
1.	the resulting strings may be very valuable for NLP processing (links to knowledge graphs);
2.	I am thinking of an extension for modeling configurable products (cars and other), and for that I would like to have a generic name already; trim is very car-specific and does not even fit for other vehicles.
3.	schema:name may include the full make and model, while configurationName will just contain strings referring to the car configuration. In European cars, this may be much more detailed than just trim information, because they are very highly configurable. The configuration string in there will be too long to be used as a short name.
4.	Yandex initially requested this property, if I remember correctly.

[Tom] If we want to keep this more general, then how about just configuration? This opens the possibility in a future extension for us to create a new type to capture additional aspects of the configuration itself, which would also be useful when modeling what products can be purchased (vs. already existing ones as are covered by the current proposal).


"The engine properties (displacement, name, power, and type) should probably be separated out into their own type, especially if we want to model multiple engines per vehicle (as some of the descriptions suggest). Otherwise, it will not be explicit which property sets go together."
--> While we could add a type Engine and set the domain of all these properties to MotorizedRoadVehicle OR Engine, and define a property "engine" with a domain of MotorizedRoadVehicle and a range of Engine, I would like to keep that as a future extension, because most cars have only one engine, and the proposal is already pretty substantial.

[Tom] I still think we should create the Engine type and make the domain of all of these properties *only* Engine, not MotorizedRoadVehicle. It doesn’t make the proposal significantly larger (just one new type), and it prevents us from having to either keep two nearly-equivalent ways of saying the same thing (using the engine property or adding directly to MotorizedRoadVehicle) or deprecating the MotorizedRoadVehicle domain for these properties when we *do* support multiple engines in the future (which, while still uncommon, is becoming much more common in the US at least). 


"For enginePower, should we also list the unit code for horsepower (BHP?)?"
--> I decided against listing it, because horsepower is to my knowledge an obsolete unit. As far as I can see, the proper code would be N12 ("Pferdestaerke", with 1 PS = 735,498 75 W, btw. It could still be used, but I think we should not actively promote it.

[Tom] Horsepower is still the most commonly used unit for engine power in the US publications I have read. I checked www.caranddriver.com and www.roadandtrack.com for some examples, and they were all in horsepower.


"For interiorType, if this is intended to describe just the material, should we rename it to interiorMaterial?"
--> I would like to keep interiorType, because it could also be a type that is not based on a material but on a target usage or audience ("Hunter edition with gun rack" or "1970s disco interior").

[Tom] OK, but can you update the description to make that clearer?


"What is the use case for productionDate? I’m not sure we need this property, but if we do, can we reuse the existing dateCreated property instead? It sounds like it would have the same meaning."
--> In Europe, the production date is often an essential characteristic for new and used car, because some cars stand on stock for several months depending on factory capacities and demand. The property was also present in the older Yandex proposal. So I suggest to keep that.

[Tom] I still don’t understand the difference vs. dateCreated. I agree it’s important to know when the car was produced, but I think that’s the same as saying it’s important to know when the car was created.


"Can we rename purchasingDate to purchaseDate and firstRegistration to firstRegisteredDate? Our date property naming in schema.org isn’t very self-consistent, but I think these suggestions will be more consistent."
--> OK, DONE: Renamed purchasingDate to purchaseDate and firstRegistration to dateOfFirstRegistration, which I think is better English.

[Tom] If we are going to put date as a prefix, how about dateFirstRegistered, to parallel the existing dateCreated, datePublished, dateModified, etc. We can leave purchaseDate as is (it matches several other existing properties that end with “Date”).

**Comment by mfhepp** [comment]
"configurationName seems to be duplicating a bunch of other properties (transmission, enginePower, bodyType, etc.). If people are going to do this, IMO, they should just use the name property. There is a concept of trim level that the US version of AutoTrader uses, which I think is missing and would be useful. That would capture probably just the “ST 2.5” part of the example. For a Subaru Legacy, on the US AutoTrader site, for example, you’ll see values such as “2.5GT Limited”, “2.5i”, “3.6R Limited”, etc. "
--> I would like to keep that, because
1. the resulting strings may be very valuable for NLP processing (links to knowledge graphs);
2. I am thinking of an extension for modeling configurable products (cars and other), and for that I would like to have a generic name already; trim is very car-specific and does not even fit for other vehicles.
3. schema:name may include the full make and model, while configurationName will just contain strings referring to the car configuration. In European cars, this may be much more detailed than just trim information, because they are very highly configurable. The configuration string in there will be too long to be used as a short name.
4. Yandex initially requested this property, if I remember correctly.

[Tom] If we want to keep this more general, then how about just configuration? This opens the possibility in a future extension for us to create a new type to capture additional aspects of the configuration itself, which would also be useful when modeling what products can be purchased (vs. already existing ones as are covered by the current proposal).

[mfhepp] - Okay, implemented in extension and  cross-checked examples (that do not use this so far). See https://github.com/mfhepp/schemaorg/commit/be853cba63c997042a446f68baedbd9cc0940624

**Comment by mfhepp** [comment]
As for:
"What is the use case for productionDate? I’m not sure we need this property, but if we do, can we reuse the existing dateCreated property instead? It sounds like it would have the same meaning."
--> In Europe, the production date is often an essential characteristic for new and used car, because some cars stand on stock for several months depending on factory capacities and demand. The property was also present in the older Yandex proposal. So I suggest to keep that.

[Tom] I still don’t understand the difference vs. dateCreated. I agree it’s important to know when the car was produced, but I think that’s the same as saying it’s important to know when the car was created.

[mfhepp]: I would stick to productionDate, because dateCreated would be very uncommon terminology in the auto industry, same as if you used dateCreated for a person's birthday (which might be biologically incorrect anyway ;-).

**Comment by mfhepp** [comment]
"For enginePower, should we also list the unit code for horsepower (BHP?)?"
--> I decided against listing it, because horsepower is to my knowledge an obsolete unit. As far as I can see, the proper code would be N12 ("Pferdestaerke", with 1 PS = 735,498 75 W, btw. It could still be used, but I think we should not actively promote it.

[Tom] Horsepower is still the most commonly used unit for engine power in the US publications I have read. I checked www.caranddriver.com and www.roadandtrack.com for some examples, and they were all in horsepower.

[mfhepp] ok, agreed. I added BHP and N12 as unit codes for brake horsepower and metric horsepower (PS). See https://github.com/mfhepp/schemaorg/commit/468335e38a2134571d42f2fb180c11300219a625

**Comment by mfhepp** [comment]
"For interiorType, if this is intended to describe just the material, should we rename it to interiorMaterial?"
--> I would like to keep interiorType, because it could also be a type that is not based on a material but on a target usage or audience ("Hunter edition with gun rack" or "1970s disco interior").

[Tom] OK, but can you update the description to make that clearer?

[mfhepp] Done. See https://github.com/mfhepp/schemaorg/commit/37aa6b70ddd6c5d6020d126e4aac843d254bfde5

**Comment by mfhepp** [comment]
"Can we rename purchasingDate to purchaseDate and firstRegistration to firstRegisteredDate? Our date property naming in schema.org isn’t very self-consistent, but I think these suggestions will be more consistent."
--> OK, DONE: Renamed purchasingDate to purchaseDate and firstRegistration to dateOfFirstRegistration, which I think is better English.

[Tom] If we are going to put date as a prefix, how about dateFirstRegistered, to parallel the existing dateCreated, datePublished, dateModified, etc. We can leave purchaseDate as is (it matches several other existing properties that end with “Date”).

[mfhepp] ok, done and cross-checked that no examples are affected. See https://github.com/mfhepp/schemaorg/commit/aa057f230a49fb90c5318f51bda9297b7680de0c

**Comment by mfhepp** [comment]
Last item on the list ;-)
"The engine properties (displacement, name, power, and type) should probably be separated out into their own type, especially if we want to model multiple engines per vehicle (as some of the descriptions suggest). Otherwise, it will not be explicit which property sets go together."
--> While we could add a type Engine and set the domain of all these properties to MotorizedRoadVehicle OR Engine, and define a property "engine" with a domain of MotorizedRoadVehicle and a range of Engine, I would like to keep that as a future extension, because most cars have only one engine, and the proposal is already pretty substantial.

[Tom] I still think we should create the Engine type and make the domain of all of these properties only Engine, not MotorizedRoadVehicle. It doesn’t make the proposal significantly larger (just one new type), and it prevents us from having to either keep two nearly-equivalent ways of saying the same thing (using the engine property or adding directly to MotorizedRoadVehicle) or deprecating the MotorizedRoadVehicle domain for these properties when we do support multiple engines in the future (which, while still uncommon, is becoming much more common in the US at least).

[mfhepp] I implemented the proposed change and think it makes the model much cleaner. See  https://github.com/mfhepp/schemaorg/commit/3f4abba2ed25efd3ed12da475efa42e592083416 for details.

Basically what I did was:

1. Change engineName to engine with a range of EngineSpecification and a domain of vehicle
2. Move all engine properties to EngineSpecification (a few went to Vehicle or EngineSpecification to cater for simpler cases)
engineDisplacement
enginePower
engineType
torque
3. deleted MotorizedRoadVehicle + all references; mostly lifted to Vehicle, sometimes Car OR BusOrCoach.
4. Updated examples in all sytaxes.

The pull request is now ready to go from my perspective. Give that this last change is non-trivial, it might be a good idea to check that I did not introduce any new bugs ;-)


But again - this is basically ready for inclusion into sdo-gozer from my point of view.

**Comment by tmarshbing** [comment]
The changes look good to me. Thanks for all the hard work, @mfhepp!
